src/pymanopt/backends/numpy_backend.py

```diff
@@ -35,6 +35,11 @@ def __init__(self, dtype: type = np.float64):
             or dtype == np.complex128
         ), f"dtype {dtype} is not supported"
         self._dtype = dtype
+        self._dtype_precision = (
+            DTypePrecision.SINGLE
+            if (dtype == np.float32 or dtype == np.complex64)
+            else DTypePrecision.DOUBLE
+        )
```

Comment:
If you dynamically change `self._dtype` then `self._dtype_precision` is not changed accordingly. I think it's better to define a getter for `self._dtype_precision` to avoid this.